app/forms/delivery_partner_form.rb

```diff
@@ -3,20 +3,22 @@
 class DeliveryPartnerForm
   include ActiveModel::Model
 
-  attr_accessor :name, :lead_providers, :provider_relationships
+  attr_accessor :name, :lead_providers, :provider_relationships, :provider_relationship_hashes
   validate :lead_providers_and_cohorts_validation
 
+  def self.provider_relationship_value(lead_provider, cohort)
+    "{\"lead_provider_id\": \"#{lead_provider.id}\", \"cohort_id\": \"#{cohort.id}\"}"
```

Comment:
i'm not entirely sure what's going on here but can you use a hash and `to_json` instead of this?

Reply:
I wasn't convinced that would guarantee order. Might be wrong though

Reply:
that's a fair point

Reply:
I don't get why this has to be a string

Reply:
https://stackoverflow.com/questions/31418673/is-order-of-a-ruby-hash-literal-guaranteed

Reply:
It's being set as the `value` of a checkbox, so needs to be a string. The issue is I'm trying to create hypothetical `ProviderRelationship` object for each of the possible combinations. They don't have IDs because they don't exist

Reply:
If we want a hypothetical relation, can we use `ProviderRelationship.new`? They still won't 'exist' in our database, but should be real enough for our code, right?

Reply:
No. How do you refer to it across controller actions? It won't be in the DB, so won't have an ID

Reply:
Something just feels wrong here. Surely, there has to be a better way. That said, I checked it and it works.

Reply:
I read that SO post as more "no one is really sure" 😂 Besides, it's not just the hash itself I'm worried about, it's the hash => json.  If you're both convinced it will never break when the serializer implementation changes, then that's good enough for me and I'll change to hash.to_json

Reply:
Sure, let's change to to_json. When it breaks in future we'll fix it and perhaps we'll have a better fix then.